js/src/components/tree-select-control/index.js

```diff
@@ -140,23 +159,29 @@ const TreeSelectControl = ( {
 		return repository;
 	}, [ treeOptions ] );
 
-	/**
+	/*
 	 * Perform the search query filter in the Tree options
 	 *
 	 * 1. Check if the search query is already cached and return it if so.
-	 * 2. Deep Copy the tree. Since we are going to modify its children and labels recursively.
+	 * 2. Deep copy the tree with adding properties for rendering.
 	 * 3. In case of filter, we apply the filter option function to the tree.
 	 * 4. In the filter function we also highlight the label with the matching letters
 	 * 5. Finally we set the cache with the obtained results and apply the filters
-	 *
 	 */
-	useEffect( () => {
+	const filteredOptions = useMemo( () => {
 		const cachedFilteredOptions = filteredOptionsCache.current[ filter ];
 
+		if ( cachedFilteredOptions ) {
+			return cachedFilteredOptions;
+		}
+
+		const isFiltered = Boolean( filter );
```

Comment:
❓  Why is this necessary? 

Just using `filter` should work I guess

Reply:
In this context, `isFiltered` is more semantic than `filter`, and the explicit boolean type allows the below return value to always be a boolean type.

https://github.com/woocommerce/google-listings-and-ads/blob/afa8de0035bd66217caa96b5212fb51ff1ea2be9/js/src/components/tree-select-control/index.js#L258-L266

Reply:
When I saw it I thought for a moment that was kind of a flag indicating that a node was filtered or not. 

💅  Maybe consider to change it to `hasFilters`

Reply:
> When I saw it I thought for a moment that was kind of a flag indicating that a node was filtered or not.

The naming was taken from the former prop used in the Options component.

https://github.com/woocommerce/google-listings-and-ads/blob/65d232a5aab91dad7910ec352a6389ff4c6f140c/js/src/components/tree-select-control/options.js#L26

> Maybe consider to change it to `hasFilters`

How about `isSearching`?

Reply:
isSearching looks pretty accurate for me yes! 🤝

Reply:
Changed in https://github.com/woocommerce/google-listings-and-ads/pull/1522/commits/6b13282a3297613e3d68cd3589315b9c6721e339.